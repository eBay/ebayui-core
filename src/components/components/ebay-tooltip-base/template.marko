<span
    w-on-expander-expand="handleExpand"
    w-on-expander-collapse="handleCollapse"
    w-body
    overlay-style=data.overlayStyle
    w-bind/>
