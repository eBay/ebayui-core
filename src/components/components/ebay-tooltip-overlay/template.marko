<script marko-init>
    var constants = require("./constants");
    var typeRoles = constants.typeRoles;
    var pointerStyles = constants.pointerStyles;
</script>

<var hasUserStyles=(
    data.styleTop ||
    data.styleLeft ||
    data.styleRight ||
    data.styleBottom
)/>
<var overlayStyles=(hasUserStyles ? {
    top: data.styleTop,
    left: data.styleLeft,
    right: data.styleRight,
    bottom: data.styleBottom
} : pointerStyles[data.pointer || "bottom"])/>

<div
    class="${data.type}__overlay"
    role=typeRoles[data.type]
    aria-labelledby=(data.type === "tourtip" && widget.elId("tourtip-label"))
    id=data.tooltipId
    style=overlayStyles
    w-bind>
    <span class="${data.type}__pointer ${data.type}__pointer--${data.pointer}"/>
    <div class="${data.type}__mask">
        <div class="${data.type}__cell">
            <div class="${data.type}__content">
                <if(data.heading)>
                    <h2
                        class=["${data.type}__heading", data.heading.class]
                        style=data.heading.style
                        w-id="tourtip-label"
                        w-body=data.heading.renderBody/>
                </if>

                <if(data.content)>
                    <div
                        body-only-if(!data.content.class && !data.content.style)
                        class=data.content.class
                        style=data.content.style
                        w-body=data.content.renderBody/>
                </if>
            </div>

            <if(data.type !== "tooltip")>
                <button
                    aria-label=data.a11yCloseText
                    class="${data.type}__close"
                    type="button"
                    w-on-click="handleCloseButton">
                    <span aria-hidden="true" role="img"/>
                </button>
            </if>
        </div>
    </div>
</div>
