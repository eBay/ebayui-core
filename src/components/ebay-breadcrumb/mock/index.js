const getItem = (text, href) => {
    const item = {
        href: 'https://www.ebay.com/',
        navSrc: '{"actionKind":"NAVSRC","operationId":"2489527"}',
        _sp: 'p2489527.m4340.l9751.c1'
    };
    const clonedItem = JSON.parse(JSON.stringify(item));
    clonedItem.renderBody = (stream) => stream.write(text);
    clonedItem.href = href;
    return clonedItem;
};
const basicItems = [getItem('eBay', 'https://www.ebay.com'),
    getItem('Auto Parts and Vehicles', 'https://www.ebay.com/b/Auto-Parts-and-Vehicles/6000/bn_1865334'),
    getItem('Motors Parts & Accessories', null)];

const firstItemMissingHref = [getItem('eBay', null),
    getItem('Auto Parts and Vehicles', 'https://www.ebay.com/b/Auto-Parts-and-Vehicles/6000/bn_1865334'),
    getItem('Motors Parts & Accessories', null)];

module.exports = {
    basicItems: {
        headingText: 'Page navigation',
        items: basicItems
    },
    hijax: {
        hijax: true,
        items: basicItems
    },
    firstItemMissingHref: {
        headingText: 'Page navigation',
        items: firstItemMissingHref
    },
    itemsWithHeadingLevel: {
        headingText: 'Page navigation',
        headingLevel: 'h3',
        items: basicItems
    }
};
