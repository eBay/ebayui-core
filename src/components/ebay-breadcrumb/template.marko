<script marko-init>
    var processHtmlAttributes = require("../../common/html-attributes");
</script>

<nav w-bind
    aria-labelledby=widget.elId("breadcrumb-heading")
    class=["breadcrumb", data.class]
    style=data.style
    role="navigation"
    ${processHtmlAttributes(data)}>

    <${data.a11yHeadingTag || "h2"} id=widget.elId("breadcrumb-heading") class="clipped">
        ${data.a11yHeadingText}
    </>
    <ul>
        <li for(item in data.items | status-var="status")>
            <var current=(!item.href && status.isLast())/>
            <${item.href ? "a" : "button"}
                href=item.href
                aria-current=(current && "location")
                class=item.class
                style=item.style
                w-body=item.renderBody
                w-onclick=(!current && "handleClick")
                ${processHtmlAttributes(item)}/>
        </li>
    </ul>
</nav>
