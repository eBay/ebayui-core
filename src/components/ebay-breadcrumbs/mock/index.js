import { createRenderBody, getNItems } from '../../../common/test-utils/shared';

export const Links = {
    a11yHeadingText: 'Page navigation',
    items: getNItems(3, (i) => ({
        href: i === 2 ? undefined : '#',
        renderBody: createRenderBody(`Item Text ${i}`),
    })),
};
export const Buttons = {
    items: getNItems(3, (i) => ({
        renderBody: createRenderBody(`Item Text ${i}`),
    })),
};
export const linkLastWithoutHREF = {
    a11yHeadingText: 'Page navigation',
    items: getNItems(3, (i) => ({
        href: i === 2 ? undefined : '#',
        renderBody: createRenderBody(`Item Text ${i}`),
    })),
};
export const linkHeadingTag = Object.assign({}, Links, {
    a11yHeadingTag: 'h3',
});
