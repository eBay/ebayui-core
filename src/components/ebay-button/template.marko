<script marko-init>
    var getWidgetId = require("../../common/get-marko-3-widget-id");
    var processHtmlAttributes = require("../../common/html-attributes");
</script>

<var size=(data.size || "medium")/>
<var priority=(data.priority || "secondary")/>
<var variant=(!data.variant && data.href ? "fake" : data.variant)/>
<var isIconVariant=(variant === 'icon')/>
<var isExpandVariant=(variant === 'expand')/>
<var isCtaVariant=(variant === 'cta')/>
<var isBadged=Boolean(data.badgeNumber && isIconVariant)/>
<var noText=(isIconVariant || isBadged || (isExpandVariant && data.noText))/>
<var baseClass=(variant ? "${variant}-btn" : "btn")/>
<var sizeClass="${baseClass}--${size}"/>
<var htmlAttributes=processHtmlAttributes(data)/>
<var tag=(data.href ? "a" : "button")/>

<${tag}
    ${htmlAttributes}
    w-bind
    w-onclick="handleClick"
    w-onkeydown="handleKeydown"
    id=(data.id || getWidgetId(out))
    class=[
        data.class,
        baseClass,
        noText && "${baseClass}--no-text",
        isBadged && "${baseClass}--badged",
        data.fluid && "${baseClass}--fluid",
        data.truncate && "${sizeClass}-truncated",
        data.fixedHeight && "${sizeClass}-fixed-height",
        !data.fixedHeight && !data.truncate && sizeClass,
        (priority === "secondary" || priority === "primary") && "${baseClass}--${priority}"
    ]
    style=data.style
    href=data.href
    type=(tag === "button" && data.type || "button")
    disabled=data.disabled
    aria-disabled=(data.partiallyDisabled && "true")>

    <if(isCtaVariant)>
        <span class="cta-btn__cell">
            <span w-body/>
            <ebay-icon type="inline" name="arrow-right-bold" class="cta-btn__icon" no-skin-classes/>
        </span>
    </if>
    <else>
        <var hasAriaLabel=Boolean(htmlAttributes.ariaLabel)/>
        <span body-only-if(!hasAriaLabel) w-body aria-hidden="true"/>
        <if(isBadged)>
            <ebay-badge
                number=data.badgeNumber
                type="icon"
                aria-label=(hasAriaLabel && data.badgeAriaLabel)
                aria-hidden=(hasAriaLabel && "true")/>
        </if>
    </else>
</>
