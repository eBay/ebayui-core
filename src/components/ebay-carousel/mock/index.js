const renderBody = (stream) => stream.write('text');
const itemWidth = 200;
const item = { renderBody, '*': { style: `width:${itemWidth}px` } };
const threeItems = [item, item, item];
const sixItems = threeItems.concat(threeItems);
const twelveItems = sixItems.concat(sixItems);

// mocks for visual debugging
const debugItem = { renderBody, '*': { style: `height:200px;width:400px;background:gray` } };
const debugItems = [debugItem, debugItem, debugItem, debugItem, debugItem, debugItem];

module.exports = { renderBody, itemWidth, item, threeItems, sixItems, twelveItems, debugItems };
