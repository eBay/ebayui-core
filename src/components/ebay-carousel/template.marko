<div class=data.classes style=data.style w-bind ${data.htmlAttributes}>
    <var config=data.config/>
    <var statusId=("carousel-status-" + widget.id)/>
    <div w-id="container" class=["carousel__container", data.bothControlsDisabled && "carousel__container--controls-disabled"]>
        <${data.a11yStatusTag}
            if(data.totalSlides >= 1)
            id=statusId
            class="clipped"
            aria-live=(data.autoplayInterval && !data.paused ? "off" : "polite")>
            <span>${data.a11yStatusText}</span>
        </>
        <button
            class="carousel__control carousel__control--prev"
            w-onclick=(!data.prevControlDisabled && "handleMove")
            data-direction=-1
            aria-describedby=statusId
            aria-label=data.a11yPreviousText
            aria-disabled=(data.prevControlDisabled && 'true')>
            <ebay-icon type="inline" name="chevron-left"/>
        </button>

        <ul w-id="list"
            class="carousel__list"
            style=(!config.nativeScrolling && data.offset && {
                transform:  "translate3d(" + (data.offset * -1) + "px,0,0)",
                transition: data.disableTransition ? 'none' : undefined
            })>
            <for(item in data.items)>
                <li ${item.htmlAttributes}
                    class=item.class
                    style=item.style
                    w-body=item.renderBody
                    aria-hidden=(!item.fullyVisible && 'true')
                    w-preserve-body-if(!!config.preserveItems)/>
            </for>
        </ul>

        <button
            w-id="next"
            class="carousel__control carousel__control--next"
            w-onclick=(!data.nextControlDisabled && "handleMove")
            data-direction=1
            aria-describedby=statusId
            aria-label=data.a11yNextText
            aria-disabled=(data.nextControlDisabled && 'true')>
            <ebay-icon type="inline" name="chevron-right"/>
        </button>

        <if(data.autoplayInterval && !data.bothControlsDisabled)>
            <if(data.paused)>
                <button
                    class="carousel__play"
                    aria-label=data.a11yPlayText
                    w-onclick="togglePlay">
                    <ebay-icon type="inline" name="play"/>
                </button>
            </if>
            <else>
                <button
                    class=`carousel__pause`
                    aria-label=data.a11yPauseText
                    w-onclick="togglePlay">
                    <ebay-icon type="inline" name="pause"/>
                </button>
            </else>
        </if>
    </div>

    <ul class="carousel__dots" if(!data.noDots && data.totalSlides > 1)>
        <li for(i from 0 to data.totalSlides - 1)>
            <var isActive=(i === data.slide)/>
            <button
                class={"carousel__dot--active": isActive}
                w-onclick=(!isActive && "handleDotClick")
                data-slide=i
                aria-describedby=statusId
                aria-disabled=(isActive && 'true')
                aria-label=(isActive ?
                    data.a11yCurrentText.replace('{currentSlide}', i + 1) :
                    data.a11yOtherText.replace('{slide}', i + 1))/>
        </li>
    </ul>
</div>
