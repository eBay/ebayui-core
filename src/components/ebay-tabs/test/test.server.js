import { describe, it } from "vitest";

import { snapshotHTML } from "../../../common/test-utils/snapshots";
import { testPassThroughAttributes } from "../../../common/test-utils/server";
import template from "../index.marko";
import * as mock from "./mock";
const htmlSnap = snapshotHTML(__dirname);

describe("tabs", () => {
    it("renders basic version with 3 tabs and 3 panels", async () => {
        const input = mock.basic3Headings_3Panels_No_Index;
        await htmlSnap(template, input);
    });

    it("renders basic version with 3 tabs and 3 panels on the second panel", async () => {
        const input = mock.basic3Headings_3Panels_1Index;
        await htmlSnap(template, input);
    });

    testPassThroughAttributes(template);
});

describe("tabs-heading", () => {
    testPassThroughAttributes(template, {
        child: {
            name: "tabs",
            multiple: true,
        },
    });
});

describe("tabs-panel", () => {
    testPassThroughAttributes(template, {
        child: {
            name: "panels",
            multiple: true,
        },
    });
});
