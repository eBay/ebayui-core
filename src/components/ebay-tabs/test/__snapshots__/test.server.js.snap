// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tabs > passes through additional html attributes 1`] = `
<div
  class="tabs class1"
  data-passed-through="true"
  data-testid="AttributePassthrough"
  style="color:red"
  type="number"
>
  <div
    class="tabs__items"
    role="tablist"
  />
  <div
    class="tabs__content"
  />
</div>
`;

exports[`tabs > passes through additional html attributes 2`] = `
<div
  class="tabs class1"
  data-passed-through="true"
  data-testid="AttributePassthrough"
  style="color:red"
  type="number"
>
  <div
    class="tabs__items"
    role="tablist"
  />
  <div
    class="tabs__content"
  />
</div>
`;

exports[`tabs > renders basic version with 3 tabs and 3 panels 1`] = `
"[36m<DocumentFragment>[39m
  [36m<div[39m
    [33mclass[39m=[32m"tabs"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"tabs__items"[39m
      [33mrole[39m=[32m"tablist"[39m
    [36m>[39m
      [36m<div[39m
        [33maria-controls[39m=[32m"s0-tabpanel-0"[39m
        [33maria-selected[39m=[32m"true"[39m
        [33mclass[39m=[32m"tabs__item"[39m
        [33mid[39m=[32m"s0-tabs-0"[39m
        [33mrole[39m=[32m"tab"[39m
      [36m>[39m
        [36m<span>[39m
          [0mHeading 0[0m
        [36m</span>[39m
      [36m</div>[39m
      [36m<div[39m
        [33maria-controls[39m=[32m"s0-tabpanel-1"[39m
        [33mclass[39m=[32m"tabs__item"[39m
        [33mid[39m=[32m"s0-tabs-1"[39m
        [33mrole[39m=[32m"tab"[39m
      [36m>[39m
        [36m<span>[39m
          [0mHeading 1[0m
        [36m</span>[39m
      [36m</div>[39m
      [36m<div[39m
        [33maria-controls[39m=[32m"s0-tabpanel-2"[39m
        [33mclass[39m=[32m"tabs__item"[39m
        [33mid[39m=[32m"s0-tabs-2"[39m
        [33mrole[39m=[32m"tab"[39m
      [36m>[39m
        [36m<span>[39m
          [0mHeading 2[0m
        [36m</span>[39m
      [36m</div>[39m
    [36m</div>[39m
    [36m<div[39m
      [33mclass[39m=[32m"tabs__content"[39m
    [36m>[39m
      [36m<div[39m
        [33maria-labelledby[39m=[32m"s0-tabs-0"[39m
        [33mclass[39m=[32m"tabs__panel"[39m
        [33mid[39m=[32m"s0-tabpanel-0"[39m
        [33mrole[39m=[32m"tabpanel"[39m
      [36m>[39m
        [36m<div[39m
          [33mclass[39m=[32m"tabs__cell"[39m
        [36m>[39m
          [36m<div>[39m
            [0mPanel 0[0m
          [36m</div>[39m
        [36m</div>[39m
      [36m</div>[39m
      [36m<div[39m
        [33maria-labelledby[39m=[32m"s0-tabs-1"[39m
        [33mclass[39m=[32m"tabs__panel"[39m
        [33mhidden[39m=[32m""[39m
        [33mid[39m=[32m"s0-tabpanel-1"[39m
        [33mrole[39m=[32m"tabpanel"[39m
      [36m>[39m
        [36m<div[39m
          [33mclass[39m=[32m"tabs__cell"[39m
        [36m>[39m
          [36m<div>[39m
            [0mPanel 1[0m
          [36m</div>[39m
        [36m</div>[39m
      [36m</div>[39m
      [36m<div[39m
        [33maria-labelledby[39m=[32m"s0-tabs-2"[39m
        [33mclass[39m=[32m"tabs__panel"[39m
        [33mhidden[39m=[32m""[39m
        [33mid[39m=[32m"s0-tabpanel-2"[39m
        [33mrole[39m=[32m"tabpanel"[39m
      [36m>[39m
        [36m<div[39m
          [33mclass[39m=[32m"tabs__cell"[39m
        [36m>[39m
          [36m<div>[39m
            [0mPanel 2[0m
          [36m</div>[39m
        [36m</div>[39m
      [36m</div>[39m
    [36m</div>[39m
  [36m</div>[39m
[36m</DocumentFragment>[39m"
`;

exports[`tabs > renders basic version with 3 tabs and 3 panels on the second panel 1`] = `
"[36m<DocumentFragment>[39m
  [36m<div[39m
    [33mclass[39m=[32m"tabs"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"tabs__items"[39m
      [33mrole[39m=[32m"tablist"[39m
    [36m>[39m
      [36m<div[39m
        [33maria-controls[39m=[32m"s0-tabpanel-0"[39m
        [33mclass[39m=[32m"tabs__item"[39m
        [33mid[39m=[32m"s0-tabs-0"[39m
        [33mrole[39m=[32m"tab"[39m
      [36m>[39m
        [36m<span>[39m
          [0mHeading 0[0m
        [36m</span>[39m
      [36m</div>[39m
      [36m<div[39m
        [33maria-controls[39m=[32m"s0-tabpanel-1"[39m
        [33maria-selected[39m=[32m"true"[39m
        [33mclass[39m=[32m"tabs__item"[39m
        [33mid[39m=[32m"s0-tabs-1"[39m
        [33mrole[39m=[32m"tab"[39m
      [36m>[39m
        [36m<span>[39m
          [0mHeading 1[0m
        [36m</span>[39m
      [36m</div>[39m
      [36m<div[39m
        [33maria-controls[39m=[32m"s0-tabpanel-2"[39m
        [33mclass[39m=[32m"tabs__item"[39m
        [33mid[39m=[32m"s0-tabs-2"[39m
        [33mrole[39m=[32m"tab"[39m
      [36m>[39m
        [36m<span>[39m
          [0mHeading 2[0m
        [36m</span>[39m
      [36m</div>[39m
    [36m</div>[39m
    [36m<div[39m
      [33mclass[39m=[32m"tabs__content"[39m
    [36m>[39m
      [36m<div[39m
        [33maria-labelledby[39m=[32m"s0-tabs-0"[39m
        [33mclass[39m=[32m"tabs__panel"[39m
        [33mhidden[39m=[32m""[39m
        [33mid[39m=[32m"s0-tabpanel-0"[39m
        [33mrole[39m=[32m"tabpanel"[39m
      [36m>[39m
        [36m<div[39m
          [33mclass[39m=[32m"tabs__cell"[39m
        [36m>[39m
          [36m<div>[39m
            [0mPanel 0[0m
          [36m</div>[39m
        [36m</div>[39m
      [36m</div>[39m
      [36m<div[39m
        [33maria-labelledby[39m=[32m"s0-tabs-1"[39m
        [33mclass[39m=[32m"tabs__panel"[39m
        [33mid[39m=[32m"s0-tabpanel-1"[39m
        [33mrole[39m=[32m"tabpanel"[39m
      [36m>[39m
        [36m<div[39m
          [33mclass[39m=[32m"tabs__cell"[39m
        [36m>[39m
          [36m<div>[39m
            [0mPanel 1[0m
          [36m</div>[39m
        [36m</div>[39m
      [36m</div>[39m
      [36m<div[39m
        [33maria-labelledby[39m=[32m"s0-tabs-2"[39m
        [33mclass[39m=[32m"tabs__panel"[39m
        [33mhidden[39m=[32m""[39m
        [33mid[39m=[32m"s0-tabpanel-2"[39m
        [33mrole[39m=[32m"tabpanel"[39m
      [36m>[39m
        [36m<div[39m
          [33mclass[39m=[32m"tabs__cell"[39m
        [36m>[39m
          [36m<div>[39m
            [0mPanel 2[0m
          [36m</div>[39m
        [36m</div>[39m
      [36m</div>[39m
    [36m</div>[39m
  [36m</div>[39m
[36m</DocumentFragment>[39m"
`;

exports[`tabs-heading > passes through additional html attributes from child tabs 1`] = `
<div
  aria-controls="s0-tabpanel-0"
  aria-selected="true"
  class="class1 tabs__item"
  data-passed-through="true"
  data-testid="AttributePassthrough"
  id="s0-tabs-0"
  role="tab"
  style="color:red"
  type="number"
>
  <span />
</div>
`;

exports[`tabs-heading > passes through additional html attributes from child tabs 2`] = `
<div
  aria-controls="s0-tabpanel-0"
  aria-selected="true"
  class="class1 tabs__item"
  data-passed-through="true"
  data-testid="AttributePassthrough"
  id="s0-tabs-0"
  role="tab"
  style="color:red"
  type="number"
>
  <span />
</div>
`;

exports[`tabs-panel > passes through additional html attributes from child panels 1`] = `
<div
  aria-labelledby="s0-tabs-0"
  class="class1 tabs__panel"
  data-passed-through="true"
  data-testid="AttributePassthrough"
  id="s0-tabpanel-0"
  role="tabpanel"
  style="color:red"
  type="number"
>
  <div
    class="tabs__cell"
  >
    <div />
  </div>
</div>
`;

exports[`tabs-panel > passes through additional html attributes from child panels 2`] = `
<div
  aria-labelledby="s0-tabs-0"
  class="class1 tabs__panel"
  data-passed-through="true"
  data-testid="AttributePassthrough"
  id="s0-tabpanel-0"
  role="tabpanel"
  style="color:red"
  type="number"
>
  <div
    class="tabs__cell"
  >
    <div />
  </div>
</div>
`;
